hopsworksai/resource_cluster.go

```diff
@@ -1316,7 +1459,7 @@ func createClusterBaseRequest(d *schema.ResourceData) (*api.CreateCluster, error
 	if v, ok := d.GetOk("ssh_key"); ok {
 		createCluster.SshKeyName = v.(string)
 	} else {
-		if _, ok := d.GetOk("aws_attributes"); !ok {
+		if _, ok := d.GetOk("azure_attributes"); ok {
```

Comment:
Shouldn't we also return an error in both cases? Azure and AWS

Reply:
no, in case of aws, adding an ssh key is optional while in case of azure it is required